### DISCUSSION TITLE
Improve perfom using flashlist  library

### DISCUSSION THREAD

**Discussion by Bound3R** [srivastavaanurag79/react-native-paper-select#66] [Ideas]
### Suggestion: Improve performance with Shopify's FlashList

Hi, 

First of all, thank you for this awesome library! 🥳 

I’d like to suggest using [FlashList](https://shopify.github.io/flash-list/) from Shopify as the default list component for handling large datasets. 
#### Why FlashList?
FlashList is an ultra-fast, highly optimized list component designed specifically for large datasets. It addresses several performance bottlenecks that often arise with React Native’s default `FlatList`. 

In my personal experience, while using `react-native-paper-select` with large datasets (for example, when working with country, department, and city lists), I’ve encountered multiple performance issues related to `FlatList`. These include slow rendering and sluggish scrolling when handling long lists. This led me to explore FlashList as a potential solution.

#### Benefits:
1. **Improved Performance:** FlashList is optimized to handle large datasets with better performance and smoother scrolling.
2. **Minimal API Changes:** The API is very similar to FlatList, so the transition could be straightforward.
3. **Support for Dynamic Lists:** Users who work with dynamically loaded lists would greatly benefit from the more efficient rendering.

#### Areas to Consider:
1. **Estimated Item Size:** One of the key aspects of FlashList is calculating the `estimatedItemSize`, which is crucial for its performance optimization. This is something that would need to be addressed, but it can be handled efficiently with some adjustments in the component.

Here's the link to the FlashList library for reference: [FlashList - Shopify](https://shopify.github.io/flash-list/).

Thanks again for your hard work, and I hope this suggestion is helpful! 😉

**Comment by srivastavaanurag79** [comment]
Working on it

**Comment by srivastavaanurag79** [comment]
[v1.3.0](https://www.npmjs.com/package/react-native-paper-select/v/1.3.0) published on npm.

- Replaced React Native FlatList with Shopify FlashList for improved performance.
- Optimized multiple functions, including list handling, item selection logic, and state management.
- Improved render efficiency by utilizing `extraData` in FlashList for better re-render control.
- Enhanced functionality for checking/unchecking items with streamlined code.
- Refactored functions for cleaner, more efficient code, reducing unnecessary operations
  and improving overall performance.
